sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1126,6 +1133,44 @@ Exceptions:
 
 |===
 
+==== New Event Member Functions
+
+Table {counter: tableNumber}. Additional member functions of the `sycl::event` class.
+[cols="2a,a"]
+|===
+|Member function|Description
+
+|
+[source,c++]
+----
+template <typename Param>
+typename Param::return_type 
+event::ext_oneapi_get_profiling_info(node Node) const;
+----
+
+| Queries the profiling information of a SYCL Graph node for the graph
+  execution associated with this SYCL event. If the requested info is
+  not available when this member function is called due to incompletion of
+  command groups associated with the event, then the call to this member
+  function will block until the requested info is available.
```

Comment:
Keeping the following aspect as a different thread: 
We also need to consider kernel/node fusion as an optimization strategy. In this case we might not have the exact profiling info.
@sommerlukas what are your thoughts on this?

Reply:
Thanks for your feedback.
I agree that it might be interesting for users to get node profiling information when using the kernel fusion feature. 
But the proposed profiling API is based on the level-zero function `zeCommandListAppendQueryKernelTimestamps` (https://spec.oneapi.io/level-zero/latest/core/api.html#zecommandlistappendquerykerneltimestamps), which in turn relies on the ZeEvents that were linked to the kernels when they were enqueued to their `command-list`. So I don't know what happens to this ZeEvents when the kernels are fused. 
Could you tell me where I can find more information on the implementation of the kernel fusion feature?

Reply:
The command-buffers that instantiate the command-list for LevelZero backend are  only created during the `command_graph<...>::finalize` process.

Reply:
I think in that case it would be a viable implementation option to first perform fusion and only then create the command buffer inside `command_graph<...>::finalize`. The command-buffer would only contain the fused kernel command (and memory commands if necessary) and it would be possible to return a `zeEvent` that could be used for profiling of the fused kernel execution.

Reply:
> I believe that profiling info for the fused kernel would be valuable for the user, so I would prefer the first option here.

I agree that profiling information is useful for a fused graph as a whole, but I don't see how it's useful to attempt to provide this profiling information on a per-node basis.  How can the application know what the profiling information means?  Won't it depend on the implementation's ability to fuse each node?

Let's consider that an application calls `ext_oneapi_get_profiling_info` on a node in a fused graph.  The returned information might tell the time it took to execute that one node (if the node could not be fused), it might tell the time it took to execute several nodes in the graph (if the node was fused with some of its neighbors), or it might tell the time it took to execute the entire graph (if all the nodes in the graph were fused).  With so much uncertainty, I don't see why this is a useful feature.

I think it does make sense to let the application get profiling information for the entire graph, but the application can do this already using the `event` that is returned from `queue::ext_oneapi_graph`.

I think `ext_oneapi_get_profiling_info` should simply fail if the graph was created with fusion enabled.  Per-node profiling information is not available for a fused graph.

Reply:
It may also be worth noting that the SYCL 2020 wording is phrased in terms of "command groups".

Getting the profiling information for the command group in which the graph was submitted, where graph execution is the "action", is consistent with the specification wording, and is always well defined.  Getting the profiling information for nodes in the graph seems less consistent to me: the mapping of nodes to command groups and "actions" is unclear.